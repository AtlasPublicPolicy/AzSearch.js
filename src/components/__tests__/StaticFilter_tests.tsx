import * as React from "react";
import * as TestUtils from "react-dom/test-utils";
import StaticFilter from "../StaticFilter";

const filters = [
    { displayName: "Any", filter: "" },
    { displayName: "House", filter: "type eq 'House'" },
    { displayName: "Apartment", filter: "type eq 'Apartment'" }
];
const defaultFilter = "";
const title = "Home Type";

describe("<StaticFilter/>", () => {
    it("renders against dummy snapshot", () => {

        expect(TestUtils.createRenderer().render(
            <StaticFilter  css={null} filters={filters} activeFilter={defaultFilter} filterKey={"type"} beforeFirstRequest={false} onFilterChange={function(foo){}} title={title} />
        )).toMatchSnapshot();
    });
});