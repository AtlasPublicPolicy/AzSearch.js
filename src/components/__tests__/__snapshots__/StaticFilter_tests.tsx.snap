// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticFilter/> renders against dummy snapshot 1`] = `
<div
  className="sorting__sortyBy panel-body"
>
  <div
    className="searchResults__facetHeader-container panel-heading"
  >
    <h4
      className="searchResults__facetHeader panel-title"
    >
      <a
        className="searchResults__facetHeader-link"
        data-toggle="collapse"
      >
        <span
          aria-hidden="true"
          className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
        />
         
        Home Type
      </a>
    </h4>
  </div>
  <div
    className="searchResults__facetControl-container panel-collapse collapse in"
  >
    <div
      className="searchResults__facetControl-list list-group"
    >
      <select
        className="sorting__sortyByControl list-group-item"
        onChange={[Function]}
      >
        <option
          selected={true}
          value=""
        >
          Any
        </option>
        <option
          selected={false}
          value="type eq 'House'"
        >
          House
        </option>
        <option
          selected={false}
          value="type eq 'Apartment'"
        >
          Apartment
        </option>
      </select>
    </div>
  </div>
</div>
`;
